### PR TITLE
feat: set lotus_info from lotus plugin at startup

### PR DESCRIPTION
**Pull request by frrist**
Makes progress towards: https://github.com/filecoin-project/sentinel/issues/34
(need to stop the Prometheus gathering via a config change over in sentinel land)

### CODE REVIEW

File: plugins/inputs/lotus/lotus.go [hunk 1]
```diff
@@ -97,20 +101,20 @@ func (l *lotus) getAPIUsingLotusConfig() (api.FullNode, func(), error) {
 		nodeCloser func()
 	)
 	if len(l.Config_APIListenAddr) > 0 && len(l.Config_APIToken) > 0 {
-		api, apiCloser, err := rpc.GetFullNodeAPIUsingCredentials(l.Config_APIListenAddr, l.Config_APIToken)
+		lotusAPI, apiCloser, err := rpc.GetFullNodeAPIUsingCredentials(l.Config_APIListenAddr, l.Config_APIToken)
```
Comment (frrist) [inline plugins/inputs/lotus/lotus.go:109]: one off: fixing an import path naming collision

File: plugins/inputs/lotus/lotus.go [hunk 2]
```diff
@@ -383,6 +396,37 @@ func recordTipsetMessagesPoints(ctx context.Context, api api.FullNode, acc teleg
...
+		return fmt.Errorf("developer error, version string has changed format")
+	}
+
+	acc.AddFields("lotus_info",
+		map[string]interface{}{
+			"recorded_at": receivedAt.UnixNano(),
+		},
+		map[string]string{
+			"api_version": version.APIVersion.String(),
+			"block_delay": fmt.Sprintf("%d", version.BlockDelay),
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go]: Probably don't need `block_delay` as it won't change really. (Unless you think there's some data display details we don't want to hardcode `30s` into our SQL queries.

File: plugins/inputs/lotus/lotus.go [hunk 3]
```diff
@@ -383,6 +396,37 @@ func recordTipsetMessagesPoints(ctx context.Context, api api.FullNode, acc teleg
...
+			"commit":      versionTokens[1],
+			"network":     string(network),
+			"peer_id":     receivedFrom.String(),
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go]: Nit: `receivedFrom` doesn't really make sense in this context. Let's just call the var `peerID`.

File: plugins/inputs/lotus/lotus.go [hunk 4]
```diff
@@ -383,6 +396,37 @@ func recordTipsetMessagesPoints(ctx context.Context, api api.FullNode, acc teleg
...
+		return err
+	}
+
+	network, err := api.StateNetworkName(ctx)
+	if err != nil {
+		return err
+	}
+	versionTokens := strings.Split(version.Version, "+")
+	if len(versionTokens) != 2 {
+		return fmt.Errorf("developer error, version string has changed format")
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go]: Let's not return an error here. If the version doesn't split, let's fill the version column with everything and leave commit blank.

File: plugins/inputs/lotus/lotus.go [hunk 5]
```diff
@@ -132,6 +136,15 @@ func (l *lotus) Start(acc telegraf.Accumulator) error {
 	}
 	l.api = nodeAPI
 
+	nodePeerID, err := l.api.ID(context.Background())
+	if err != nil {
+		return err
+	}
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go:146]: Push the nodePeerID getting into `recordLotusInfoPoints`

File: plugins/inputs/lotus/lotus.go [hunk 6]
```diff
@@ -132,6 +136,15 @@ func (l *lotus) Start(acc telegraf.Accumulator) error {
...
+
+	if err := recordLotusInfoPoints(context.Background(), l.api, acc, time.Now(), nodePeerID); err != nil {
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go]: Push `time.Now()` into the function and remove as argument, along with nodePeerID.

**Review by placer14 — CHANGES_REQUESTED** [review]
Small set of changes needed.

File: plugins/inputs/lotus/lotus.go (hunk 5 above)
Comment (frrist) [reply plugins/inputs/lotus/lotus.go:146]: :+1: done

File: plugins/inputs/lotus/lotus.go [hunk 7]
```diff
@@ -464,6 +469,47 @@ func recordTipsetMessagesPoints(ctx context.Context, api api.FullNode, acc teleg
...
+		version = versionTokens[0]
+		commit = versionTokens[1]
+	} else {
+		log.Println("W! developer error, version string has changed format")
+		version = v.Version
+	}
+
+	acc.AddFields("lotus_info",
+		map[string]interface{}{
+			"recorded_at": time.Now().UnixNano(),
```
Comment (placer14) [inline plugins/inputs/lotus/lotus.go]: Sorry I overlooked this the first time. A timestamp is automatically associated with each `AddFields` call. This fields will double the data we capture for little extra value.

**Review by placer14 — APPROVED** [review]
GTM w one minor issue. I'll fix and merge. Thanks!

**Pull request merged**